clippy_lints/src/loops/manual_memcpy.rs

```diff
@@ -63,8 +64,9 @@ pub(super) fn check<'tcx>(
                             && get_slice_like_element_ty(cx, cx.typeck_results().expr_ty(base_right)).is_some()
                             && let Some((start_left, offset_left)) = get_details_from_idx(cx, idx_left, &starts)
                             && let Some((start_right, offset_right)) = get_details_from_idx(cx, idx_right, &starts)
-
-                            // Source and destination must be different
+							&& !local_used_in(cx, canonical_id, base_left)
+							&& !local_used_in(cx, canonical_id, base_right)
+                        // Source and destination must be different
                             && path_to_local(base_left) != path_to_local(base_right)
```

Comment:
Comment needs reindenting, otherwise looks good to me. A squash of the commits would be nice

Reply:
I fixed the indentation and squashed the commits into one.
Thank you for your sincere guidance!